projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java

```diff
@@ -57,7 +64,17 @@ public void updateJobSecrets(String jobName, Map<String, Object> secrets) {
                       return entry.getValue();
                     }));
 
-    vaultJobSecrets.setSecretsJson(new JSONObject(updatedSecrets).toString());
+    String updatedSecretsString = new JSONObject(updatedSecrets).toString();
+    try {
+      if (updatedSecretsString.getBytes("UTF-8").length > sizeLimit) {
+        throw new DataJobSecretsSizeLimitException(
+            jobName, "Secret size exceeds configured limit of:" + sizeLimit);
+      }
+    } catch (UnsupportedEncodingException e) {
+      throw new DataJobSecretsException(jobName, e.getMessage());
```

Comment:
nit: e.toString() might be more helpful as it include the class name.

Reply:
fixed.